joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForFunctionsCreator.scala

```diff
@@ -45,14 +45,14 @@ trait AstForFunctionsCreator(implicit withSchemaValidation: ValidationMode) { th
     val name     = funcDecl.json(ParserKeys.Name).obj(ParserKeys.Name).str
     val fullname = s"$fullyQualifiedPackage.$name"
     // TODO: handle multiple return type or tuple (int, int)
+    val genericTypeMethodMap = processTypeParams(funcDecl.json(ParserKeys.Type))
     val (returnTypeStr, methodReturn) =
-      getReturnType(funcDecl.json(ParserKeys.Type)).headOption
+      getReturnType(funcDecl.json(ParserKeys.Type), genericTypeMethodMap).headOption
         .getOrElse(("", methodReturnNode(funcDecl, Defines.voidTypeName)))
-    val templateParams       = ""
-    val params               = funcDecl.json(ParserKeys.Type)(ParserKeys.Params)(ParserKeys.List)
-    val genericTypeMethodMap = processTypeParams(funcDecl.json(ParserKeys.Type))
+    val templateParams = ""
+    val params         = funcDecl.json(ParserKeys.Type)(ParserKeys.Params)(ParserKeys.List)
     val signature =
-      s"$fullname$templateParams(${parameterSignature(params, genericTypeMethodMap)})$returnTypeStr"
+      s"$fullname$templateParams(${parameterSignature(params, genericTypeMethodMap)})($returnTypeStr)"
```

Comment:
`(`$returnTypeStr`)` this addition of bracket will break many unit tests inside MethodTest

Reply:
@pandurangpatil should i change all those unit test?

Reply:
change the unit tests

Reply:
@ankit-privado remove the brackets and update only your unit tests

Reply:
okay sure @pandurangpatil